test/integration/http_subset_lb_integration_test.cc

```diff
@@ -176,7 +176,10 @@ class HttpSubsetLbIntegrationTest
       }
     }
 
-    if (is_hash_lb_) {
+    // The default number of choices for the LEAST_REQUEST policy is 2 hosts, if the number of hosts
+    // is equal to the number of choices, a full scan happens instead, this means that the same host
+    // will be chosen.
+    if (is_hash_lb_ || (GetParam() == envoy::config::cluster::v3::Cluster::LEAST_REQUEST)) {
```

Comment:
If there is an way of checking the number of hosts of the cluster so we can make sure the test has the correct behaviour that would be great, for now I added a comment on why we need this check on the test here.